src/sdk/pynni/nni/tuner.py

```diff
@@ -93,3 +93,16 @@ def _on_exit(self):
 
     def _on_error(self):
         pass
+
+    def extract_scalar_reward(self, value, scalar_key='default'):
+        if isinstance(value, float) or isinstance(value, int):
+            value = value
+        elif isinstance(value, dict) and scalar_key in value:
+            value = value[scalar_key]
+            if isinstance(value, float) or isinstance(value, int):
+                pass
+            else:
+                raise RuntimeError('Incorrect final result: the final result for %s should be float/int, or a dict which has a key named "default" whose value is float/int.' % str(self.__class__))
+        else:
+            raise RuntimeError('Incorrect final result: the final result for %s should be float/int, or a dict which has a key named "default" whose value is float/int.' % str(self.__class__)) 
+        return value
```

Comment:
if you throw error anyway, how about change the code as below:
if isinstance(value, dict):
    value = value[scalar_key]
if isinstance(value, float) or instance(value, int):
    return value
else:
    raise ....